### ISSUE TITLE
Fail to compile v1.18.37

### ISSUE THREAD

**Issue by psycofdj** [cloudfoundry/backup-and-restore-sdk-release#612]
Hello,

My CI has trouble compiling backup-and-restore-sdk-release `v1.18.37` against default stemcell ~~xenial/1.x~~ `ubuntu-bionic/1.x`

I got the following error:
```
+ go build -mod vendor -o /var/vcap/packages/s3-blobstore-backup-restorer/bin/s3-blobstore-backup-restore ./cmd/s3-blobstore-backup-restore
go: inconsistent vendoring in /var/vcap/data/compile/s3-blobstore-backup-restorer/s3-blobstore-backup-restore:
	s3-blobstore-backup-restore: is marked as replaced in vendor/modules.txt, but not replaced in go.mod

	To ignore the vendor directory, use -mod=readonly or -mod=mod.
	To sync the vendor directory, run:
		go mod vendor
```

**Comment by fejnartal** [comment]
Hello @psycofdj 
Thanks for reporting this.

My team and I are some of the maintainers of this project.
Apparently, you also tested this release against xenial stemcell (not only bionic) and it was having trouble to compile too.
We will be prioritising fixing compilation with xenial.

Adding support for bionic is not among the highest priorities of my team at the moment.
Could you contribute a PR to make supporting bionic stemcell possible?
We would be very keen to review any PR implementing such functionality
after we have fixed the compilation issues with xenial.

In any case, we have already added this to our backlog.

**Comment by psycofdj** [comment]
Hello

My first comment was a mistake. I haven't test compilation against xenial and can only confirm the bug on bionic.
I'm very surprised that you focus on xenial since:
- this distribution has reached its end of support since April 2021
- bionic has become the default stemcell in all major deployments (bosh, cf, concourse)

For the record, my team and I (and many other ppl in the community) are already flawlessly using this bosh-release compiled against bionic for almost a year.

In all case, I'm 99% sure that the underlying problem has nothing to do with the stemcell but is simply a matter of golang version in interaction with:
- either the vendored packages in [./src/s3-blobstore-backup-restore](https://github.com/cloudfoundry/backup-and-restore-sdk-release/tree/main/src/s3-blobstore-backup-restore)
- either the command line used to compile the binaries [here](https://github.com/cloudfoundry/backup-and-restore-sdk-release/blob/main/packages/s3-blobstore-backup-restorer/packaging#L25)

I could not reliably determine the version of the golang use in the compilation process since its package is vendored in the bosh release. I could not reproduce the compilation bug in my local environment.

**Comment by psycofdj** [comment]
The regression was introduced in the following commit f435f3608f77958b4e47cddb0d60bc15746bbbc5

My theory is that the new included golang version has a stricter check of `go.mod` vs `modules.txt` (as described in the error message). The fix should be simply to resync the vendor directory against `go.mod` 

Question: why does the module replaces itself in the `go.mod` ? 
https://github.com/cloudfoundry/backup-and-restore-sdk-release/blob/e20a4fec7963057f8e7aa2e2bd1de065909f4e62/src/s3-blobstore-backup-restore/go.mod#L1
https://github.com/cloudfoundry/backup-and-restore-sdk-release/blob/e20a4fec7963057f8e7aa2e2bd1de065909f4e62/src/s3-blobstore-backup-restore/go.mod#L15

**Comment by dlresende** [comment]
Hi @psycofdj. @fejnartal and I just released [1.18.38](https://github.com/cloudfoundry/backup-and-restore-sdk-release/releases/tag/v1.18.38). We've tested the Golang compilation issue was fixed against Xenial. Let us know if it also works with Bionic.

**Comment by psycofdj** [comment]
I confirm that [1.18.38](https://github.com/cloudfoundry/backup-and-restore-sdk-release/releases/tag/v1.18.38) is now :100: on ubuntu-bionic/1.x, thank you for the `go.mod` fixes
